apps/internal/oauth/oauth.go

```diff
@@ -194,14 +195,28 @@ func (d DeviceCode) Token(ctx context.Context) (accesstokens.TokenResponse, erro
 	}
 }
 
-var waitRE = regexp.MustCompile("(authorization_pending|slow_down)")
+type DeviceCodeError struct {
```

Comment:
Do we expect callers outside the `oauth` package to need this type?  If not then should not be exported.

Reply:
Changed it to not be exported in this commit https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/178/commits/5fbd566d1d01afddf610abb63fa29e05fff314ce